### PR TITLE
py-parsl: add v2023.08.21

### PR DESCRIPTION
**Pull request by myan-rarcellino-nnl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-parsl/package.py [hunk 1]
```diff
@@ -16,16 +16,21 @@ class PyParsl(PythonPackage):
...
 
-    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.8:", type=("build", "run"), when="@2023.08.21:")
+    depends_on("python@3.6:", type=("build", "run"), when="@:1.2")
     depends_on("py-setuptools", type="build")
     depends_on("py-pyzmq@17.1.2:", type=("build", "run"))
-    depends_on("py-typeguard@2.10:", type=("build", "run"))
+    depends_on("py-typeguard@2.10:2", type=("build", "run"), when="@2023.08.21:")
+    depends_on("py-typeguard@2.10:", type=("build", "run"), when="@:1.2")
     depends_on("py-typing-extensions", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-parsl/package.py]: Has version reqs in the latest version

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-parsl/package.py]: Missing six dep

File: var/spack/repos/builtin/packages/py-parsl/package.py [hunk 2]
```diff
@@ -35,13 +40,16 @@ class PyParsl(PythonPackage):
     depends_on("py-psutil@5.5.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-parsl/package.py:42]: Missing setproctitle dep

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-parsl/package.py (hunk 1 above)
Comment (myan-rarcellino-nnl) [reply var/spack/repos/builtin/packages/py-parsl/package.py]: not sure how i missed these, but thanks for checking

File: var/spack/repos/builtin/packages/py-parsl/package.py (hunk 2 above)
Comment (myan-rarcellino-nnl) [reply var/spack/repos/builtin/packages/py-parsl/package.py:42]: added

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Will give @hategan a chance to review before merging

**Comment by hategan** [comment]
Thanks. Looks fine to me.

**Review by hategan — APPROVED** [review]

**Pull request merged**